mypy/semanal.py

```diff
@@ -2366,11 +2374,19 @@ def fail_typeddict_arg(self, message: str,
 
     def build_typeddict_typeinfo(self, name: str, items: List[str],
                                  types: List[Type]) -> TypeInfo:
-        mapping_value_type = join.join_type_list(types)
         fallback = (self.named_type_or_none('typing.Mapping',
```

Comment:
The reason why we don't use a mutable type as a fallback is that it would compromise type safety. For example, if the fallback would be `Dict[str, (join of value types)]`, then for `{'a': 1, 'b': 'x'}` the fallback type would be `Dict[str, object]`. Through the fallback type we could do things like `del d['a']` or `d['a'] = [1]` which would break safety. `Mapping` only supports getting values, so it's safe. Also, since `Mapping` only provides read operations, it can be covariant in the value type.

Making a typed dict compatible with `Dict[str, Any]` without making this the fallback might be a reasonable thing to do, but it's unclear if that would result in ambiguity or other problems.